segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/StreamSegmentInformation.java

```diff
@@ -46,24 +53,30 @@
     /**
      * Creates a new instance of the StreamSegmentInformation class.
      *
-     * @param name         The name of the StreamSegment.
-     * @param startOffset  The first available offset in this StreamSegment.
-     * @param length       The length of the StreamSegment.
-     * @param sealed       Whether the StreamSegment is sealed (for modifications).
-     * @param deleted      Whether the StreamSegment is deleted (does not exist).
-     * @param attributes   The attributes of this StreamSegment.
-     * @param lastModified The last time the StreamSegment was modified.
+     * @param name             The name of the StreamSegment.
+     * @param startOffset      The first available offset in this StreamSegment.
+     * @param length           The length of the StreamSegment.
+     * @param sealed           Whether the StreamSegment is sealed (for modifications).
+     * @param deleted          Whether the StreamSegment is deleted (does not exist).
+     * @param storageLength    Storage length.
+     * @param sealedInStorage  Whether the StreamSegment is sealed (for modifications) in storage.
+     * @param deletedInStorage Whether the StreamSegment is deleted (does not exist) in storage.
+     * @param attributes       The attributes of this StreamSegment.
+     * @param lastModified     The last time the StreamSegment was modified.
      */
     @Builder
-    private StreamSegmentInformation(String name, long startOffset, long length, boolean sealed, boolean deleted,
-                                    Map<UUID, Long> attributes, ImmutableDate lastModified) {
+    private StreamSegmentInformation(String name, long startOffset, long length, long storageLength, boolean sealed, boolean deleted,
+                                     boolean sealedInStorage, boolean deletedInStorage, Map<UUID, Long> attributes, ImmutableDate lastModified) {
         Preconditions.checkArgument(startOffset >= 0, "startOffset must be a non-negative number.");
         Preconditions.checkArgument(length >= startOffset, "length must be a non-negative number and greater than startOffset.");
         this.name = Exceptions.checkNotNullOrEmpty(name, "name");
         this.startOffset = startOffset;
         this.length = length;
+        this.storageLength = storageLength;
         this.sealed = sealed;
+        this.sealedInStorage = sealedInStorage;
         this.deleted = deleted;
+        this.deletedInStorage = deletedInStorage;
```

Comment:
Precondition check. If `deletedInStorage == true`, then `deleted` must be true as well.

Reply:
Updated